src/tds/codec/column_data.rs

```diff
@@ -502,12 +502,16 @@ impl<'a> ColumnData<'a> {
         let data = Self::decode_plp_type(src, mode).await?;
 
         let res = if let Some(buf) = data {
-            if buf.len() % 2 != 0 {
-                return Err(Error::Protocol("nvarchar: invalid plp length".into()));
-            }
+            if ty == VarLenType::BigChar {
+                Some(String::from_utf8(buf)?)
```

Comment:
So these are always utf-8 encoded, and nchar is utf-16?

Reply:
No. Char is ASCII. In MSSQL 2019 it can store UTF-8 also.